scripts/src/analyzeMessages.ts

```diff
@@ -0,0 +1,133 @@
+import Path from "path";
+import { promises as fs } from "fs";
+import { MongoClient, Db } from "mongodb";
+import { Conversation, Message } from "chat-server";
+import { makeTypeChatJsonTranslateFunc, assertEnvVars } from "chat-core";
+import { MessageAnalysis } from "./MessageAnalysis";
+
+import "dotenv/config";
+
+const {
+  MONGODB_DATABASE_NAME,
+  MONGODB_CONNECTION_URI,
+  OPENAI_ENDPOINT,
+  OPENAI_API_KEY,
+  OPENAI_CHAT_COMPLETION_MODEL_VERSION,
+  OPENAI_CHAT_COMPLETION_DEPLOYMENT,
+} = assertEnvVars({
+  MONGODB_DATABASE_NAME: "",
+  MONGODB_CONNECTION_URI: "",
+  OPENAI_ENDPOINT: "",
+  OPENAI_API_KEY: "",
+  OPENAI_CHAT_COMPLETION_MODEL_VERSION: "",
+  OPENAI_CHAT_COMPLETION_DEPLOYMENT: "",
+});
+
+async function main() {
+  const client = await MongoClient.connect(MONGODB_CONNECTION_URI);
+  try {
+    const db = client.db(MONGODB_DATABASE_NAME);
+    await analyzeMessages({ db });
+  } finally {
+    await client.close();
+  }
+}
+
+main();
+
+const analyzeMessages = async ({ db }: { db: Db }) => {
+  const conversationsCollection = db.collection<Conversation>("conversations");
+  const originalMessages = conversationsCollection.aggregate<Message>([
+    {
+      // Find messages in a recent timeframe
+      $match: {
+        "messages.createdAt": {
+          // Originals for older conversations are likely to have been deleted,
+          // so we can't use it to get the topic
+          $gte: daysBeforeDate(8),
+        },
+      },
+    },
+    {
+      // With replaceRoot below, pass each message from conversation to next
+      // stage in pipeline
+      $unwind: "$messages",
+    },
+    {
+      $replaceRoot: { newRoot: "$messages" },
+    },
+    {
+      // Filter out non-user messages
+      $match: {
+        role: "user",
+      },
+    },
+    {
+      // Join with the scrubbed collection in order to find those that need
+      // analysis
+      $lookup: {
+        from: "scrubbed_messages",
+        localField: "id",
+        foreignField: "_id",
+        as: "scrubbed",
+      },
+    },
+    {
+      // Filter out messages that haven't been scrubbed yet or that already have
+      // analysis set
+      $match: {
+        scrubbed: { $ne: [] },
```

Comment:
```suggestion
        "scrubbed.0": { $exists: true },
```

Don't think it's really functionally different but the array index pattern is also an efficient way to check a minimum array size.